otherlibs/systhreads/st_stubs.c

```diff
@@ -675,6 +712,10 @@ CAMLprim value caml_thread_exit(value unit)   /* ML */
 {
   struct longjmp_buffer * exit_buf = NULL;
 
+  /* we check if another domain was ever started */
+  if (caml_domain_is_multicore())
+    caml_fatal_error("Thread.exit: unsupported call under multiple domains");
```

Comment:
OK, but I'd rather call `caml_invalid_argument` so that the whole program is not aborted and the current thread gets a chance to react to the error (finalization).